File: backend/src/main/java/ru/sportexpress/menuApp/repository/MenuRepository.java
package ru.sportexpress.menuApp.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import ru.sportexpress.menuApp.entity.Menu;

//@N1qlPrimaryIndexed
//@ViewIndexed(designDoc = "menu", viewName = "all")
public interface MenuRepository extends JpaRepository<Menu, Integer> {

}

File: backend/src/main/kotlin/ru/sportexpress/menuApp/repository/MenuRepository.kt
package ru.sportexpress.menuApp.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import ru.sportexpress.menuApp.entity.Menu


interface MenuRepository : JpaRepository<Menu, Long> {

//    @Query("SELECT m FROM Menu m WHERE m.parentId is Null")
    fun findAllByParentIdIsNull(): List<Menu>
}
